### DISCUSSION TITLE
protocol WM_DELETE_WINDOW

### DISCUSSION THREAD

**Discussion by paitonero** [TomSchimansky/CustomTkinter#1407] [Q&A]
Hello all.
I am trying to control the close window event:

root.protocol("WM_DELETE_WINDOW, on_closing)

def on_closing():
     if messagebox.askokcancel(--------)
         root.destroy()
The problem I have is that the code does not work.
I read that it is neccessary to use a tkinter window and I am working with a customertkinter one, perhaps I can not control this event if I'm using a customertkinter window? or should I follow a different way if I am working with customertkinter?

Thanks for help

**Comment by avalon60** [comment]
Hi. I use this in one of my CustomTkinter apps to prevent a window being closed:
```

    self.preview.protocol("WM_DELETE_WINDOW", self.block_closing)

    def block_closing(self, event=0):
        pass

```

This works for me.

**Reply by paitonero** [reply]
Thanks for your answer.
AT end it works for me as follows:

def on_closing(ventana):
    if messagebox.askyesno(message= "¿Desea salir de la aplicación?", title= "Aviso salir Aplicación"):
        bi.conx.clienteConex.close()    # Aprovecho para cerrar la B.D.
        ventana.destroy()   # Destruyo la ventana principal y se termina la app.
    else:
        ventana.iconify()   # Se minimiza la ventana principal
